sdn/controller/controller.py

```diff
@@ -47,4 +47,23 @@ def add_logical_port(self, p):
         p.container_ip = container_ip
         p.underlay_network_ip = self.get_network(p.network.id).ip
         self.router.add_logical_port(p)
-        p.container.add_logical_port(p)
\ No newline at end of file
+        p.container.add_logical_port(p)
+
+    def remove_logical_port(self, net_id, container_name):
+        logger.info("Removing logical port id: %s, ip: %s", net_id, container_name)
+
+        logger.debug("Notifying router and container")
+        lp = self.router.get_logical_port(container_name, net_id)
+        lp.container.remove_logical_port(lp)
+        self.router.remove_logical_port(lp)
+
+        logger.debug("Removing docker networks")
+        docker_net = self.docker_client.networks.get(net_id)
+        for container in docker_net.containers:
+            docker_net.disconnect(container)
+        docker_net.remove()
```

Comment:
zwróć uwagę, że wszystkie segmenty sieci underlay mają tę samą nazwę, taką samą jak nazwa sieci overlay - mam wrażenie, że problem może być tutaj - usuwasz segment sieci, ale to czasem może być nie ten segment, który masz na myśli